contract/test/CashRemunerationEIP712.spec.ts

```diff
@@ -0,0 +1,216 @@
+import { ethers, upgrades } from 'hardhat'
+import { expect } from 'chai'
+import { SignerWithAddress } from '@nomicfoundation/hardhat-ethers/signers'
+import { CashRemunerationEIP712 } from '../typechain-types'
+
+describe('CashRemuneration (EIP712)', () => {
+  let cashRemunerationProxy: CashRemunerationEIP712
+
+  const deployContract = async (employer: SignerWithAddress) => {
+    const CashRemunerationImplementation = await ethers.getContractFactory('CashRemunerationEIP712')
+    cashRemunerationProxy = (await upgrades.deployProxy(
+      CashRemunerationImplementation,
+      [employer.address],
+      { initializer: 'initialize' }
+    )) as unknown as CashRemunerationEIP712
+  }
+
+  describe('As CNC Company Founder', () => {
+    let employer: SignerWithAddress
+    let employee: SignerWithAddress
+    let imposter: SignerWithAddress
```

Comment:
Haha, imposter kind of takes me back to the "Among Us" days.

Reply:
Haha had to look that one up